### PR TITLE
Converts loop and callees in SSLInitServerContext to std::string

### PR DESCRIPTION
**Pull request by randall**

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
I'm not sure how to clarify this as requested, it seems obvious to me. But I'll try. Consider the original code:
```
completeServerCertPath = Layout::relative_to(params->serverCertPathOnly, certname);
```
because `completeServerCertpath` is of type `ats_scoped_str`, the string currently held in `completeServerCertPath` is freed (via `ats_free`) before the new value is assigned. Creating a new instance of `ats_scoped_str` simply shifts point of releasing the memory to the end of the loop, rather than at the assignment. `ats_scoped_str` was designed specifically to make this case not leak.

**Comment by randall** [comment]
Apologies for the noise. This was detected as a leak on 7.1.x using Xcode's leak detector. Though running the same unmodified 8.0.x code, this code does not show up as a leak.

Is it possible this didn't work as expected on 7.1.x?

**Comment by zwoop** [comment]
So Do we close this?

**Comment by SolidWallOfCode** [comment]
It shouldn't have been a leak in 7.1.x either - `ats_scoped_str` didn't change for 8.x. However, `Layout::relative_to` changed to return a `std::string` instead of the result of `ats_strdup`. Looking at this, this code should be changed to use `std::string` instead of `ats_scoped_str` and do it as a local variable inside the loop. That would be more efficient.

File: iocore/net/SSLUtils.cc [hunk 1]
```diff
@@ -1686,7 +1685,7 @@ SSLInitServerContext(const SSLConfigParams *params, const ssl_user_config *sslMu
       }
 
       for (const char *certname = cert_tok.getNext(); certname; certname = cert_tok.getNext()) {
-        completeServerCertPath = Layout::relative_to(params->serverCertPathOnly, certname);
+        ats_scoped_str completeServerCertPath(Layout::relative_to(params->serverCertPathOnly, certname));
```
Comment (SolidWallOfCode) [inline iocore/net/SSLUtils.cc]: Declare this as `std:string` - that would avoid an unnecessary string duplication.

**Comment by SolidWallOfCode** [comment]
Thinking more, I know @dragon512 said the changes he, @macisasandwich, and @chitianhao made in the layout logic cleaned up some memory leaks, maybe this was one of them.

**Comment by chitianhao** [comment]
Yeah, `ats_scoped_str` shouldn't have leaks anymore with the new layout class. But those `ats_scoped_str` should be replaced by `std::string` someday.

**Comment by randall** [comment]
I've updated the subject to reflect reality. There will need to be a modified version of this patch for 7.1.x

**Comment by randall** [comment]
I think so.

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**